src/recipes/ai-context/index.ts

```diff
@@ -48,78 +70,171 @@ const promptForPath = async (): Promise<string> => {
   ])
 
   if (customPath) {
-    return customPath
+    return { ...rulesForDefaultConsumer, path: customPath || rulesForDefaultConsumer.path }
   }
 
   log('You must select a path.')
 
-  return promptForPath()
+  return promptForContextConsumerSelection()
+}
+
+/**
+ * Checks if a command belongs to a known IDEs by checking if it includes a specific string.
+ * For example, the command that starts windsurf looks something like "/applications/windsurf.app/contents/...".
+ */
+const getConsumerKeyFromCommand = (command: string): string | null => {
+  // The actual command is something like "/applications/windsurf.app/contents/...", but we are only looking for windsurf
+  const match = cliContextConsumers.find(
+    (consumer) => consumer.consumerProcessCmd && command.includes(consumer.consumerProcessCmd),
+  )
+  return match ? match.key : null
+}
+
+/**
+ * Receives a process ID (pid) and returns both the command that the process was run with and its parent process ID. If the process is a known IDE, also returns information about that IDE.
+ */
+const getCommandAndParentPID = async (
+  pid: number,
+): Promise<{
+  parentPID: number
+  command: string
+  consumerKey: string | null
+}> => {
+  const { stdout } = await execa('ps', ['-p', String(pid), '-o', 'ppid=,comm='])
```

Comment:
What happens on Windows?

Reply:
I don't know, @smnh will need to verify. Currently, this work is all wrapped in try/catch and will skip if any errors are thrown and, if it simply doesn't work, then it will also continue as it does without this logic.

Reply:
@eduardoboucas on Windows and on systems that do not support `ps` it will fallback to IDE selection.
We can add later detection for OS and run appropriate command that lists processes with their IDs and commands.